mongo/single_result.go

```diff
@@ -71,6 +95,7 @@ func (sr *SingleResult) setRdrContents() error {
 		return nil
 	case sr.cur != nil:
 		defer sr.cur.Close(context.TODO())
+
 		if !sr.cur.Next(context.TODO()) {
```

Comment:
Not part of your PR, but it's weird that we use `context.TODO()` here. We should replace those with either `context.Background()` to maintain the behavior of `context.TODO()` or use the context provided to `FindOne` in a subsequent PR.

Reply:
Great point. This also happens in `topology.go`. Filed GODRIVER-2256.